import/webview/rawwebview.cpp

```diff
@@ -14,31 +14,89 @@
 #include "webengine.h"
 #include "webenginesettings.h"
 
+#include <qmozviewcreator.h>
+
 #include <QtGui/QGuiApplication>
 #include <QtGui/QScreen>
 #include <QtGui/QStyleHints>
 #include <QtGui/QMouseEvent>
 #include <QtQuick/QQuickWindow>
 #include <private/qquickwindow_p.h>
 
+#include <algorithm>
+
 #define CONTENT_ORIENTATION_CHANGED QLatin1String("embed:contentOrientationChanged")
 
 namespace SailfishOS {
 
 namespace WebView {
 
+class ViewCreator : public QMozViewCreator
+{
+public:
+    ViewCreator();
+    ~ViewCreator();
+
+    quint32 createView(const QString &url, const quint32 &parentId) override;
+
+    static std::shared_ptr<ViewCreator> instance();
+
+    std::vector<RawWebView *> views;
```

Comment:
I think this should be set. having id and RawWebView pointer.

```std::set<quint32, RawWebView *>```

 Views without parent should be distinguishable from the views that are child views. A child view is a view that can be only closed from JavaScript as well whereas the parent view cannot.

Parent id is always zero (0). Views created from web content will always have > 0 id.

Reply:
The id would be the view's uniqueID though and that's not going to be available until the view is constructed which doesn't happen until sometime after the RawWebView constructor executes which means we'd have to notified when the view is constructed and only add the RawWebView to the global list of instances then.